openfermioncirq/optimization/black_box.py

```diff
@@ -46,8 +46,21 @@ class BlackBox(metaclass=abc.ABCMeta):
             a list of tuples of the form (low, high), where low and high are
             lower and upper bounds on a parameter. The number of tuples
             should be equal to the dimension of the black box.
+        cost_of_evaluate: If specified, then calls to `evaluate` will be
+            redirected to `evaluate_with_cost` with the specified cost.
     """
 
+    def __init__(self,
+                 cost_of_evaluate: Optional[float]=None,
+                 **kwargs) -> None:
+        """
+        Args:
+            cost_of_evaluate: An optional cost associated with the
```

Comment:
I'm sorry but I'm confused what this "cost" is meant to quantify. The docs here don't make it clear either. What is the units of the float, cost_of_evaluate?

Reply:
It doesn't make sense to talk about the units except in a concrete implementation of the BlackBox. I can give you an example though. BlackBoxes created by HamiltonianVariationalStudy have a noise and cost model in which the noise comes from the fact that you can only measure the energy a finite number of times, but performing more measurements will reduce your noise. In that situation, the cost represents the number of independent energy measurements that are performed.

Reply:
ok, I think I get it. Go ahead with this PR